src/OrchardCore/OrchardCore.ContentLocalization.Abstractions/Records/LocalizedContentItemIndex.cs

```diff
@@ -48,29 +79,28 @@ public override Task RemovedAsync(RemoveContentContext context)
         public void Describe(DescribeContext<ContentItem> context)
         {
             context.For<LocalizedContentItemIndex>()
+                .When(contentItem => contentItem.Has<LocalizationPart>() || _partRemoved.Contains(contentItem.ContentItemId))
                 .Map(contentItem =>
                 {
-                    var part = contentItem.As<LocalizationPart>();
-
-                    if (part == null)
+                    // If the content item was removed, a record is still added.
+                    if (!contentItem.Published && !contentItem.Latest && !_itemRemoved.Contains(contentItem))
```

Comment:
seems ok to do this here, as when returning to published, or republishing, the `LocalizationPart` would still contain the value, so index recreated on save.

but also maybe not, if ever a query was wanted for deleted items as part of a localization set?

Reply:
Here we do the same as for `AutoroutePartIndex` as we refresh the entries from the index table and by using a cursor, so as for autoroutes we need to add a record when deleting an item (or part).

Reply:
For the part remove, here we set Culture to null, as we set the Path to null for autoroutes

> but also maybe not, if ever a query was wanted for deleted items as part of a localization set?

That would be ok because here (as for autoroutes) we keep the record when it was deleted, and i take care that it is kept in the table, so that other running instances don't miss it

Need to leave, will see this night / tomorrow